http/base/common/common.go

```diff
@@ -521,3 +527,88 @@ func BuildNeoVMParam(builder *neovm.ParamsBuilder, smartContractParams []interfa
 	}
 	return nil
 }
+
+func PreExecCheck(txn *types.Transaction) error {
+	if _, overflow := common.SafeMul(txn.GasLimit, txn.GasPrice); overflow {
+		return fmt.Errorf("Tx GasPrice %d and gasLimit %d safeMul overflow",
+			txn.GasPrice, txn.GasLimit)
+	}
+	if txn.TxType == types.Deploy && txn.GasLimit < svrneovm.CONTRACT_CREATE_GAS {
+		return fmt.Errorf("Deploy tx gasLimit should >= %d",
+			svrneovm.CONTRACT_CREATE_GAS)
+	}
+	gasLimitConfig := config.DefConfig.Common.GasLimit
+	gasPriceConfig := getGasPriceConfig()
```

Comment:
i think you need cache this value and  refresh it  when it is expired, instead of fetching it every time.

Reply:
given consensus node will disable pre-execute, performance is not crucial for client node, so keep it like this.